homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,232 @@
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["cameraeyes"],
+            )
+        )
+
+    for switch in session.device_helper.camera_360:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["camera360"],
+            )
+        )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class SHCSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC switch."""
+
+    entity_description: SHCSwitchEntityDescription
+
+    def __init__(
+        self,
+        device: SHCDevice,
+        parent_id: str,
+        entry_id: str,
+        description: SHCSwitchEntityDescription,
+    ) -> None:
+        """Initialize a SHC switch."""
+        super().__init__(device, parent_id, entry_id)
+        self.entity_description = description
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return (
+            getattr(self._device, self.entity_description.on_key)
+            == self.entity_description.on_value
+        )
+
+    @property
+    def today_energy_kwh(self) -> float | None:
+        """Return the total energy usage in kWh."""
+        if self.entity_description.has_consumption:
+            return self._device.energyconsumption / 1000.0
+        return None
+
+    @property
+    def current_power_w(self) -> float | None:
+        """Return the current power usage in W."""
+        if self.entity_description.has_consumption:
+            return self._device.powerconsumption
+        return None
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the switch on."""
+        setattr(self._device, self.entity_description.on_key, True)
+
+    def turn_off(self, **kwargs) -> None:
+        """Turn the switch off."""
+        setattr(self._device, self.entity_description.on_key, False)
+
+    def toggle(self, **kwargs) -> None:
+        """Toggle the switch."""
+        setattr(self._device, self.entity_description.on_key, not self.is_on)
+
+    @property
+    def should_poll(self) -> bool:
+        """Switch needs polling."""
+        return self.entity_description.should_poll
+
+    def update(self) -> None:
+        """Trigger an update of the device."""
+        self._device.update()
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return the state attributes."""
+        if self.entity_description.key == "smartplug":
+            return {
+                "routing": self._device.routing.name,
+            }
+        if self.entity_description.key == "smartplugcompact":
+            return {
+                "communication_quality": self._device.communicationquality.name,
+            }
+        return None
```

Comment:
This looks like it should be a sensor entity, possibly with `DEVICE_CLASS_SIGNAL_STRENGTH`

Reply:
The states reported for communicationquality are of qualatative nature, i.e. GOOD, MEDIUM, BAD, UNKNOWN. Would you anyway suggest to create a sensor entity?

Reply:
Created a separate sensor entity.